### PR TITLE
[HOLD] Add missing @material-ui/core exports

### PR DESCRIPTION
**Pull request by drwpow**
Need to hold until we have wildcard support. But this is another usecase where auto-expanding doesn‘t apply here

### CODE REVIEW

File: exports/@material-ui/core/4.10.1.json [hunk 1]
```diff
@@ -113,5 +113,9 @@
   "./Tooltip": "./esm/Tooltip/Tooltip.js",
   "./Typography": "./esm/Typography/Typography.js",
   "./Unstable_TrapFocus": "./esm/Unstable_TrapFocus/Unstable_TrapFocus.js",
-  "./Zoom": "./esm/Zoom/Zoom.js"
+  "./Zoom": "./esm/Zoom/Zoom.js",
```
Comment (drwpow) [inline exports/@material-ui/core/4.10.1.json:116]: 🤔  would this be valid syntax?

```json
  "./*": "./esm/*/*.js"
```

If so this package would be suuuuuper easy to type 😛

File: exports/@material-ui/core/4.10.1.json [hunk 2]
```diff
@@ -113,5 +113,9 @@
...
+  "./styles/*": "./styles/*.js",
+  "./utils/*": "./utils/*.js",
+  "./package.json": "./package.json",
+  "./": "./"
```
Comment (FredKSchott) [inline exports/@material-ui/core/4.10.1.json:120]: we can't add "./"! That tells the builder that every file is the package is accessible and an entrypoint, and if every file is an entrypoint then there's no optimization to be had.

File: exports/@material-ui/core/4.10.1.json [hunk 3]
```diff
@@ -113,5 +113,9 @@
   "./Tooltip": "./esm/Tooltip/Tooltip.js",
   "./Typography": "./esm/Typography/Typography.js",
   "./Unstable_TrapFocus": "./esm/Unstable_TrapFocus/Unstable_TrapFocus.js",
-  "./Zoom": "./esm/Zoom/Zoom.js"
+  "./Zoom": "./esm/Zoom/Zoom.js",
+  "./styles/*": "./styles/*.js",
```
Comment (FredKSchott) [inline exports/@material-ui/core/4.10.1.json:117]: are you sure about this? I don't think "./styles/*" is meant to be imported. In the documentation I only see:

```
import { styled } from '@material-ui/core/styles';
```

which would mean this line should be:

```
"./styles": "./esm/styles/index.js"
```

see https://material-ui.com/styles/basics/

File: exports/@material-ui/core/4.10.1.json [hunk 4]
```diff
@@ -113,5 +113,9 @@
...
+  "./utils/*": "./utils/*.js",
```
Comment (FredKSchott) [inline exports/@material-ui/core/4.10.1.json:118]: same here, I think only this is supported:

```
"./utils": "./esm/utils/index.js"
```

Comment (drwpow) [reply exports/@material-ui/core/4.10.1.json:118]: In a few of the `esm/` component files they import other styles and utils directly other than index. I think the entire folder is meant to be imported

Comment (drwpow) [reply exports/@material-ui/core/4.10.1.json:118]: Unless in our export maps do we only need to export “public” exports? And while preparing the package anything goes?

File: exports/@material-ui/core/4.10.1.json (hunk 2 above)
Comment (drwpow) [reply exports/@material-ui/core/4.10.1.json:120]: I took that from Preact. I thought it was shallow? i.e. only expose files in root?

**Pull request closed without merging**